proof/archive.go

```diff
@@ -266,37 +286,105 @@ func NewFileArchiver(dirName string) (*FileArchiver, error) {
 	}, nil
 }
 
-// genProofFilePath generates the full proof file path based on a rootPath and
-// a valid locator. The final path is: root/assetID/scriptKey.assetproof
-func genProofFilePath(rootPath string, loc Locator) (string, error) {
-	var emptyKey btcec.PublicKey
-
+// genProofFileStoragePath generates the full proof file path for storing a
+// proof based on a rootPath and a valid locator.
+// The final path is: root/assetID/scriptKey-outpoint.assetproof.
+//
+// NOTE: Because some operating systems have issues with paths longer than 256
+// characters, we don't use the full outpoint in the file name, but only the
+// first 16 bytes (32 hex characters) of the hash. That should be enough to
+// avoid collisions but saves us a full 32 characters (we already use 130 for
+// the hex encoded asset ID and script key).
+func genProofFileStoragePath(rootPath string, loc Locator) (string, error) {
```

Comment:
Re the root path, don't we also need to consider that if we're worrying about a 260-ish char limit? In other words, not sure if we can 100% ensure an issue w.r.t never happens. 

From a cursory search, looks like FAT32 file systems would be affected, old NTFS, older linux/mac.

Reply:
Hmm, good point. Though I'm not sure we want to have logic in our code to detect if this would be an issue? And from what I can tell the error message one would get would be pretty clear (path or file name too long).
By truncating the outpoint our path is around 184 characters long (including the `proofs/` prefix) which leaves at least 66 characters for the folder name this would be located in, which should be enough. Without the truncation it would only be 34 characters which wouldn't be enough for `c:\users\guggero\appdata\roaming\tapd` for example (which is 38 characters).